File: LES_EXAMENS/EXAMEN_PYTHON_2025_2026_IA/exercice4.c
#include <stdio.h>
#include <stdbool.h>
float T[40];
float temp;
bool verifier;
int main(){
    for(int i=0;i<40;i++){
        do{
            printf("saisir la note numero %d ",i+1);
            scanf("%f",&T[i]);
            if(T[i]<0 || T[i]>20){
                printf("la note %f pas entre 0 et 20!.",T[i]);

            }
        }while(T[i]<0 || T[i]>20);
    }
    do{
        verifier=true;
        for(int i=0;i<39;i++){
            if(T[i]<T[i+1]){
                temp=T[i];
                T[i]=T[i+1];
                T[i+1]=temp;
                verifier=false;
            }
        }
    }while(verifier==false);
    for(int i=0;i<40;i++){
        printf("la note numero %d est %f \n",i+1,T[i]);
        if(T[i]>=16){
            printf("la mention Tres bien\n");
        }
        else if(T[i]>=14){
            printf("la mention bien\n");
        }
        else if(T[i]>=12){
            printf("la mention Assez bien\n");
        }
        else if(T[i]>=10){
            printf("la mention passable\n");
        }
        else{
            printf("la mention Ajourne\n");
        }
    }
    return 0;
}

File: LES_EXAMENS/EXAMEN_PYTHON_2025_2026_IA/exercice4.py
T=[None for i in range(40)]
for i in range(0,40):
    while True:
        T[i]=float(input(f"saisir la note {i+1} entre 0 et 20: "))
        if (T[i]>=0 and T[i]<=20) :
            break
        else:
            print(f"la note {T[i]} pas entre 0 et 20!.")
verifier=True
while True:
    for i in range(0,39):
        if T[i]<T[i+1]:
            verifier=False
    if verifier==False:
        for i in range(0,39):
            if T[i]<T[i+1]:
                temp=T[i]
                T[i]=T[i+1]
                T[i+1]=temp
        verifier=True
    else:
        break
for i in range(0,40):
    print(f"la note numero {i+1}: {T[i]}.")
    if T[i]>=16:
        print("la mention Tres bien.")
    elif T[i]>=14:
        print("la mention  bien.")
    elif T[i]>=12:
        print("la mention Assez bien.")
    elif T[i]>=10:
        print("la mention Passable.")
    else:
        print("la mention Ajourne.")